x-pack/plugins/ml/server/shared_services/shared_services.ts

```diff
@@ -71,7 +77,10 @@ export function createSharedServices(
   getClusterClient: () => IClusterClient | null,
   getInternalSavedObjectsClient: () => SavedObjectsClientContract | null,
   isMlReady: () => Promise<void>
-): SharedServices {
+): {
+  shared: SharedServices;
```

Comment:
as with the comment above, these could be called `sharedServices` and `internalServices` 
Also dropping the `Provider` part of the name, unless we want to add that to both.

Reply:
yes, good point, and I reckon both of them should contain `Providers` postfix because they are not actual service instances